test/test_distributed.py

```diff
@@ -2094,6 +2102,50 @@ def _join_and_reduce(self, fn):
     class TestMPI(TestCase, _DistTestBase):
         pass
 
+elif BACKEND == "test":
+    class TestBackendDynamicLoad(TestCase):
+        def setUp(self):
+            super(TestBackendDynamicLoad, self).setUp() 
+
+        def _load_test_backend(self):
+            temp_dir = tempfile.mkdtemp()
+            src = "{}/cpp_extensions/cpp_c10d_extension.cpp".format(os.path.abspath(os.path.dirname(__file__)))
+            extension = torch.utils.cpp_extension.load(
+                name="torch_test",
+                sources=[src],
+                build_directory=temp_dir
+            ) 
+
+        def test_backend_apis(self):
+            import torch.utils.cpp_extension
+            try:
+                torch.utils.cpp_extension.verify_ninja_availability()
+            except RuntimeError:
+                print(CPP_EXTENSIONS_WARNING)
+                return 0
```

Comment:
You can turn this into a decorator so you can skip the test before running it. Then the test will show up as skipped if ninja is not available. If you return from the test itself, it shows up as successful.

Reply:
done